`OidcClientRegistrationTest.testRegisteredClientDynamicTenant` in `integration-tests/oidc-client-registration/` fails on main

`OidcClientRegistrationTest.testRegisteredClientDynamicTenant` in `integration-tests/oidc-client-registration/` fails on main 

Fails since #41866 has been merged (tested against 33e1e1f8e1db22941df2dfd56c5f7a955b2617ec).

/cc @sberyozkin

> One thing that happens to me is that I forget to rebuild Quakrus after I revert changes. I suggest to try that.

That's what I did.

```bash
./mvnw -Dquickly -T32 && clear && ./mvnw verify -Dtest-containers -Dstart-containers -pl integration-tests/oidc-client-registration/
```

against 613cdf3670f2ef234c93805e5a8aacb5f09f20d0

```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running io.quarkus.it.keycloak.OidcClientRegistrationTest
2024-09-07 19:39:52,990 INFO  [org.tes.DockerClientFactory] (build-57) Testcontainers version: 1.20.1
2024-09-07 19:39:53,743 INFO  [org.tes.doc.DockerClientProviderStrategy] (build-57) Found Docker environment with Environment variables, system properties and defaults. Resolved dockerHost=unix:///run/user/1000/podman/podman.sock
2024-09-07 19:39:53,744 INFO  [org.tes.DockerClientFactory] (build-57) Docker host IP address is localhost
2024-09-07 19:39:54,227 INFO  [org.tes.DockerClientFactory] (build-57) Connected to docker: 
  Server Version: 4.9.5
  API Version: 1.41
  Operating System: ubuntu
  Total Memory: 128562 MB
2024-09-07 19:39:54,233 INFO  [org.tes.ima.PullPolicy] (build-57) Image pull policy will be performed by: DefaultPullPolicy()
2024-09-07 19:39:54,233 INFO  [org.tes.uti.ImageNameSubstitutor] (build-57) Image name substitution will be performed by: DefaultImageNameSubstitutor (composite of 'ConfigurationFileImageNameSubstitutor' and 'PrefixingImageNameSubstitutor')
2024-09-07 19:39:54,235 INFO  [org.tes.DockerClientFactory] (build-57) Checking the system...
2024-09-07 19:39:54,235 INFO  [org.tes.DockerClientFactory] (build-57) ✔︎ Docker server version should be at least 1.6.0
2024-09-07 19:39:54,346 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (build-57) Using Quarkus powered Keycloak distribution
2024-09-07 19:39:54,486 INFO  [tc.qua.io/.0.4] (build-57) Creating container for image: quay.io/keycloak/keycloak:25.0.4
2024-09-07 19:39:54,503 INFO  [tc.tes.8.1] (build-57) Creating container for image: testcontainers/ryuk:0.8.1
2024-09-07 19:39:54,541 INFO  [tc.tes.8.1] (build-57) Container testcontainers/ryuk:0.8.1 is starting: f979765deb4eb0f145ea83cece4fb3f32905672b65de8cbf02064de50c15fc6f
2024-09-07 19:39:55,305 INFO  [tc.tes.8.1] (build-57) Container testcontainers/ryuk:0.8.1 started in PT0.801551329S
2024-09-07 19:39:55,701 INFO  [tc.qua.io/.0.4] (build-57) Container quay.io/keycloak/keycloak:25.0.4 is starting: ad06d3eabda17e5a8a65b0fd29c982b8dc162b16a4d6af8092702faa8f5654c4
2024-09-07 19:39:56,646 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: Changes detected in configuration. Updating the server image.
2024-09-07 19:39:56,646 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: Updating the configuration and installing your custom providers, if any. Please wait.
2024-09-07 19:39:59,145 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: 2024-09-07 17:39:58,830 INFO  [io.qua.dep.QuarkusAugmentor] (main) Quarkus augmentation completed in 2206ms
2024-09-07 19:39:59,145 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: Server configuration updated and persisted. Run the following command to review the configuration:
2024-09-07 19:39:59,145 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: 
2024-09-07 19:39:59,146 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: 	kc.sh show-config
2024-09-07 19:39:59,146 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: 
2024-09-07 19:39:59,146 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: Next time you run the server, just run:
2024-09-07 19:39:59,146 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: 
2024-09-07 19:39:59,146 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: 	kc.sh start --http-enabled=true --hostname-strict=false --spi-user-profile-declarative-user-profile-config-file=/opt/keycloak/upconfig.json --optimized
2024-09-07 19:39:59,146 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: 
2024-09-07 19:40:01,260 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: 2024-09-07 17:40:00,950 INFO  [org.infinispan.CONTAINER] (ForkJoinPool.commonPool-worker-1) ISPN000556: Starting user marshaller 'org.infinispan.jboss.marshalling.core.JBossUserMarshaller'
2024-09-07 19:40:01,260 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: 2024-09-07 17:40:01,122 WARN  [org.jgroups.stack.Configurator] (ForkJoinPool.commonPool-worker-1) JGRP000014: ThreadPool.thread_dumps_threshold has been deprecated: ignored
2024-09-07 19:40:01,260 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: 2024-09-07 17:40:01,126 INFO  [org.infinispan.CLUSTER] (ForkJoinPool.commonPool-worker-1) ISPN000078: Starting JGroups channel `ISPN` with stack `udp`
2024-09-07 19:40:01,261 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: 2024-09-07 17:40:01,127 INFO  [org.jgroups.JChannel] (ForkJoinPool.commonPool-worker-1) local_addr: 98191a2b-8746-46ce-82e9-f2f4dd9ca350, name: ad06d3eabda1-33686
2024-09-07 19:40:01,261 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: 2024-09-07 17:40:01,129 WARN  [org.jgroups.protocols.UDP] (ForkJoinPool.commonPool-worker-1) JGRP000015: the send buffer of socket MulticastSocket was set to 1MB, but the OS only allocated 212.99KB
2024-09-07 19:40:01,261 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: 2024-09-07 17:40:01,130 WARN  [org.jgroups.protocols.UDP] (ForkJoinPool.commonPool-worker-1) JGRP000015: the receive buffer of socket MulticastSocket was set to 20MB, but the OS only allocated 212.99KB
2024-09-07 19:40:01,261 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: 2024-09-07 17:40:01,130 WARN  [org.jgroups.protocols.UDP] (ForkJoinPool.commonPool-worker-1) JGRP000015: the send buffer of socket MulticastSocket was set to 1MB, but the OS only allocated 212.99KB
2024-09-07 19:40:01,261 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: 2024-09-07 17:40:01,130 WARN  [org.jgroups.protocols.UDP] (ForkJoinPool.commonPool-worker-1) JGRP000015: the receive buffer of socket MulticastSocket was set to 25MB, but the OS only allocated 212.99KB
2024-09-07 19:40:01,261 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: 2024-09-07 17:40:01,133 INFO  [org.jgroups.protocols.FD_SOCK2] (ForkJoinPool.commonPool-worker-1) server listening on *.23406
2024-09-07 19:40:01,261 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: 2024-09-07 17:40:01,259 INFO  [org.keycloak.broker.provider.AbstractIdentityProviderMapper] (main) Registering class org.keycloak.broker.provider.mappersync.ConfigSyncEventListener
2024-09-07 19:40:03,395 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: 2024-09-07 17:40:03,135 INFO  [org.jgroups.protocols.pbcast.GMS] (ForkJoinPool.commonPool-worker-1) ad06d3eabda1-33686: no members discovered after 2001 ms: creating cluster as coordinator
2024-09-07 19:40:03,395 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: 2024-09-07 17:40:03,139 INFO  [org.infinispan.CLUSTER] (ForkJoinPool.commonPool-worker-1) ISPN000094: Received new cluster view for channel ISPN: [ad06d3eabda1-33686|0] (1) [ad06d3eabda1-33686]
2024-09-07 19:40:03,396 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: 2024-09-07 17:40:03,174 INFO  [org.infinispan.CLUSTER] (ForkJoinPool.commonPool-worker-1) ISPN000079: Channel `ISPN` local address is `ad06d3eabda1-33686`, physical addresses are `[10.88.0.126:38942]`
2024-09-07 19:40:03,396 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: 2024-09-07 17:40:03,344 INFO  [org.keycloak.connections.infinispan.DefaultInfinispanConnectionProviderFactory] (main) Node name: ad06d3eabda1-33686, Site name: null
2024-09-07 19:40:04,145 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: 2024-09-07 17:40:03,720 INFO  [org.keycloak.quarkus.runtime.storage.legacy.liquibase.QuarkusJpaUpdaterProvider] (main) Initializing database schema. Using changelog META-INF/jpa-changelog-master.xml
2024-09-07 19:40:04,895 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: 
2024-09-07 19:40:04,895 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: UPDATE SUMMARY
2024-09-07 19:40:04,896 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: Run:                        134
2024-09-07 19:40:04,896 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: Previously run:               0
2024-09-07 19:40:04,896 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: Filtered out:                 0
2024-09-07 19:40:04,896 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: -------------------------------
2024-09-07 19:40:04,896 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: Total change sets:          134
2024-09-07 19:40:04,896 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: 
2024-09-07 19:40:04,896 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: 2024-09-07 17:40:04,698 INFO  [org.keycloak.services] (main) KC-SERVICES0050: Initializing master realm
2024-09-07 19:40:05,895 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: 2024-09-07 17:40:05,505 INFO  [org.keycloak.services] (main) KC-SERVICES0009: Added user 'admin' to realm 'master'
2024-09-07 19:40:05,895 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: 2024-09-07 17:40:05,553 INFO  [io.quarkus] (main) Keycloak 25.0.4 on JVM (powered by Quarkus 3.8.5) started in 6.487s. Listening on: http://0.0.0.0:8080. Management interface listening on http://0.0.0.0:9000.
2024-09-07 19:40:05,895 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: 2024-09-07 17:40:05,553 INFO  [io.quarkus] (main) Profile prod activated. 
2024-09-07 19:40:05,896 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: 2024-09-07 17:40:05,553 INFO  [io.quarkus] (main) Installed features: [agroal, cdi, hibernate-orm, jdbc-h2, keycloak, logging-gelf, narayana-jta, reactive-routes, resteasy-reactive, resteasy-reactive-jackson, smallrye-context-propagation, vertx]
2024-09-07 19:40:05,896 INFO  [tc.qua.io/.0.4] (build-57) Container quay.io/keycloak/keycloak:25.0.4 started in PT11.409825659S
2024-09-07 19:40:06,896 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (build-57) Dev Services for Keycloak started.
2024-09-07 19:40:07,145 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: 2024-09-07 17:40:06,679 WARN  [org.keycloak.models.utils.RepresentationToModel] (executor-thread-1) Using deprecated 'credentials' format in JSON representation for user 'admin'. It will be removed in future versions
2024-09-07 19:40:07,146 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: 2024-09-07 17:40:06,685 WARN  [org.keycloak.models.utils.RepresentationToModel] (executor-thread-1) Using deprecated 'credentials' format in JSON representation for user 'alice'. It will be removed in future versions
2024-09-07 19:40:07,146 INFO  [io.qua.oid.dep.dev.key.KeycloakDevServicesProcessor] (docker-java-stream-1601966854) Keycloak: 2024-09-07 17:40:06,688 WARN  [org.keycloak.models.utils.RepresentationToModel] (executor-thread-1) Using deprecated 'credentials' format in JSON representation for user 'jdoe'. It will be removed in future versions
2024-09-07 19:40:07,375 ERROR [io.qua.oid.cli.reg.run.OidcClientRegistrationImpl] (vert.x-eventloop-thread-1) Client registeration has failed:  status: 403, error message: {"error":"insufficient_scope","error_description":"Policy 'Trusted Hosts' rejected request to client-registration service. Details: Host not trusted."}
2024-09-07 19:40:07,376 ERROR [io.qua.oid.cli.reg.run.OidcClientRegistrationRecorder] (vert.x-eventloop-thread-1) Default client registartion failed: {"error":"insufficient_scope","error_description":"Policy 'Trusted Hosts' rejected request to client-registration service. Details: Host not trusted."}, it can be retried later
2024-09-07 19:40:07,387 DEBUG [io.qua.oid.run.OidcRecorder] (main) 'Default' tenant configuration is disabled
[ERROR] Tests run: 6, Failures: 0, Errors: 1, Skipped: 5, Time elapsed: 16.00 s <<< FAILURE! -- in io.quarkus.it.keycloak.OidcClientRegistrationTest
[ERROR] io.quarkus.it.keycloak.OidcClientRegistrationTest.testRegisteredClientDynamicTenant -- Time elapsed: 0.003 s <<< ERROR!
java.lang.RuntimeException: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.test.junit.QuarkusTestExtension.throwBootFailureException(QuarkusTestExtension.java:632)
	at io.quarkus.test.junit.QuarkusTestExtension.interceptTestClassConstructor(QuarkusTestExtension.java:716)
	at java.base/java.util.Optional.orElseGet(Optional.java:364)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
Caused by: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
	at io.quarkus.runtime.Application.start(Application.java:101)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at io.quarkus.runner.bootstrap.StartupActionImpl.run(StartupActionImpl.java:305)
	at io.quarkus.test.junit.QuarkusTestExtension.doJavaStart(QuarkusTestExtension.java:245)
	at io.quarkus.test.junit.QuarkusTestExtension.ensureStarted(QuarkusTestExtension.java:599)
	at io.quarkus.test.junit.QuarkusTestExtension.beforeAll(QuarkusTestExtension.java:649)
	... 1 more
Caused by: java.lang.NullPointerException: Cannot invoke "io.quarkus.oidc.client.registration.RegisteredClient.metadata()" because "this.defaultRegClientOnStartup" is null
	at io.quarkus.it.keycloak.CustomTenantConfigResolver.onStartup(CustomTenantConfigResolver.java:53)
	at io.quarkus.it.keycloak.CustomTenantConfigResolver_Observer_onStartup_un1RJSxnxCXBc4apKBq6hVnbpKM.notify(Unknown Source)
	at io.quarkus.arc.impl.EventImpl$Notifier.notifyObservers(EventImpl.java:351)
	at io.quarkus.arc.impl.EventImpl$Notifier.notify(EventImpl.java:333)
	at io.quarkus.arc.impl.EventImpl.fire(EventImpl.java:80)
	at io.quarkus.arc.runtime.ArcRecorder.fireLifecycleEvent(ArcRecorder.java:156)
	at io.quarkus.arc.runtime.ArcRecorder.handleLifecycleEvents(ArcRecorder.java:107)
	at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent1144526294.deploy_0(Unknown Source)
	at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent1144526294.deploy(Unknown Source)
	... 8 more

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   OidcClientRegistrationTest.testRegisteredClientDynamicTenant » Runtime java.lang.RuntimeException: Failed to start quarkus
[INFO] 
[ERROR] Tests run: 6, Failures: 0, Errors: 1, Skipped: 5
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  20.984 s
[INFO] Finished at: 2024-09-07T19:40:07+02:00
[INFO] ------------------------------------------------------------------------
[INFO] 11 goals, 9 executed, 2 from cache
```

--- a/integration-tests/oidc-client-registration/src/main/resources/quarkus-realm.json
+++ b/integration-tests/oidc-client-registration/src/main/resources/quarkus-realm.json
@@ -1202,24 +1202,6 @@
   "components": {
     "org.keycloak.services.clientregistration.policy.ClientRegistrationPolicy": [
       {
-        "id": "9b4e5b69-1d07-489b-b8a5-07329c957141",
-        "name": "Trusted Hosts",
-        "providerId": "trusted-hosts",
-        "subType": "anonymous",
-        "subComponents": {},
-        "config": {
-          "host-sending-registration-request-must-match": [
-            "true"
-          ],
-          "trusted-hosts": [
-            "172.17.0.1"
-          ],
-          "client-uris-must-match": [
-            "false"
-          ]
-        }
-      },
-      {
         "id": "e2f513d3-44e3-435c-8b2a-68a5d384fd97",
         "name": "Full Scope Disabled",
         "providerId": "scope",

